src/agent0/core/hyperdrive/interactive/hyperdrive_agent.py

```diff
@@ -144,85 +148,138 @@ def add_funds(
             The amount of base to fund the agent with. Defaults to 0.
         eth: FixedPoint | None, optional
             The amount of ETH to fund the agent with. Defaults to 0.
+        pool: LocalHyperdrive | None, optional
+            The pool to interact with. Defaults to the active pool.
         signer_account: LocalAccount | None, optional
             The signer account to use to call `mint`. Defaults to the agent itself.
         """
+
+        if pool is None and self._active_pool is not None:
```

Comment:
I think there is a situation where both `pool` and `self._active_pool` are `None`. The user initializes with `pool=None` (allowed), and then calls `add_funds` with `pool=None` (allowed). But this combo would break, correct?

Reply:
This gets checked when we attempt to mint base